### PR TITLE
Move most of decoupling logic to processors

### PR DESCRIPTION
**Pull request by kwalcock**

### CODE REVIEW

**Comment by kwalcock** [comment]
These changes accompany https://github.com/clulab/processors/pull/477.  That is, if we are allowed to change processors in order to support this new reach functionality, this is how I'd do it just now.

Processors has been updated to 8.3.3-SNAPSHOT here, so this can't be used directly until the processors change is merged and processors is republished.

The second commit includes a manual reordering of the label priorities.  This needs to be fixed with a design or convention change of some kind.  I'll defer at least temporarily to others on that.

**Comment by kwalcock** [comment]
@enoriega, so you know among other things how that map was turned into a sequence.

File: processors/src/main/scala/org/clulab/processors/bionlp/ner/KBGenerator.scala [hunk 1]
```diff
@@ -98,59 +95,16 @@ object KBGenerator {
...
-      .filter(_.nonEmpty)
-      .sorted
-      .distinct
+  def processKBFiles(caseInsensitiveMatching: Boolean): Seq[StandardKbSource] = {
+    val kbEntries = loadFromConf()
+    // This should only be performed on small collections.
+    // It is important to keep both the labels and the KbEntries in the same order.
+    val actualLabels = kbEntries.map(_.neLabel).distinct
+    val expectedLabels = Seq("Gene_or_gene_product", "Family", "Cellular_component", "Simple_chemical", "Site", "BioProcess", "Disease", "Species", "CellLine", "TissueType", "CellType", "Organ")
+    val kbEntriesSeq = expectedLabels.map { label => kbEntries.filter(_.neLabel == label) }
```
Comment (enoriega) [inline processors/src/main/scala/org/clulab/processors/bionlp/ner/KBGenerator.scala:104]: This line and line 103 should go away for two reasons:
1. `loadFromConf` will guarantee the entries come in the order specified by the user in the configuration
2. A new KB file with a previously unseen label can be added by a user

I am sorry for leading you to port the hard-coded iteration order

**Review by enoriega — CHANGES_REQUESTED** [review]
Looks very nice! Just left one simple request on `KBGenerator`

File: processors/src/main/scala/org/clulab/processors/bionlp/ner/KBGenerator.scala (hunk 1 above)
Comment (kwalcock) [reply processors/src/main/scala/org/clulab/processors/bionlp/ner/KBGenerator.scala:104]: I understand and expected 103, but what is "This line"?  I'm missing the antecedent.

Comment (kwalcock) [reply processors/src/main/scala/org/clulab/processors/bionlp/ner/KBGenerator.scala:104]: Yes, thanks, that line would use the actualLabels again.

**Comment by kwalcock** [comment]
These won't pass until processors has been republished.

**Comment by kwalcock** [comment]
These are passing now with the updated version of processors, @enoriega.

**Comment by enoriega** [comment]
Thanks @kwalcock !

**Pull request merged**